pkg/controller/atlasinstance/atlasinstance_controller.go

```diff
@@ -303,6 +303,10 @@ func (r *MongoDBAtlasInstanceReconciler) reconcileAtlasProject(cx context.Contex
 		}
 		// Fetch the project name from newly created CR
 		atlasProject, err = r.getAtlasProject(cx, inst)
+
+		if err == nil && atlasProject == nil {
+			err = errors.New("could not find AtlasProject custom resource")
```

Comment:
adding return here WDYT?

Reply:
This part is refactored.

Reply:
Okay, I see you are using here Naked returned interesting - https://go.dev/tour/basics/7